### PR TITLE
SOOP_BA remove cleanup tmp dir when error during processing zip

### PR DESCRIPTION
**Pull request by bpasquer**
@lbesnard  another bit of code to fix https://github.com/aodn/data-services/issues/395#issuecomment-211175218

### CODE REVIEW

File: SOOP/SOOP_BA/incoming_handler.sh [hunk 1]
```diff
@@ -71,14 +71,25 @@ main() {
 
     local tmp_dir=`mktemp -d`
     chmod a+rx $tmp_dir
-    unzip -q -u -o $file -d $tmp_dir || file_error "Error unzipping"
+    local tmp_zip_manifest=`mktemp`
+
+    unzip_file $file $tmp_dir $tmp_zip_manifest
+    if [ $? -ne 0 ]; then
+        rm -rf --preserve-root $tmp_dir
```
Comment (lbesnard) [inline SOOP/SOOP_BA/incoming_handler.sh]: why not adding a ```rm $tmp_zip_manifest``` as well ?

File: SOOP/SOOP_BA/incoming_handler.sh [hunk 2]
```diff
@@ -71,14 +71,25 @@ main() {
...
+        file_error "Error unzipping"
+    fi
 
     local nc_file
-    nc_file=`find $tmp_dir -name "*.nc" | head -1` || file_error "Cannot find NetCDF file in zip bundle"
+    nc_file=`grep ".*.nc" $tmp_zip_manifest | head -1`
+    if [ $? -ne 0 ]; then
+        rm -rf --preserve-root $tmp_dir
```
Comment (lbesnard) [inline SOOP/SOOP_BA/incoming_handler.sh]: same here

Comment (bpasquer) [inline SOOP/SOOP_BA/incoming_handler.sh]: fixed  both

**Comment by julian1** [comment]
Why is rm -rf necessary in these cases when unzip has failed?

**Pull request merged**